src/reducer.ts

```diff
@@ -0,0 +1,122 @@
+import { calendarState, action as actionI } from "./interfaces";
+
+export const calendarReducer = (
+  state: calendarState,
+  action: actionI
+): calendarState => {
```

Comment:
I would recommend you to define the reducers as part of a folder for each one of them, and implement a helper library with the mutations you are running on the state. 

For example: 
```
case "CHANGE_CALENDAR_MODE": {
    
   // this mutation modifies the localstorage and generates the new state.
    return helper.changeCalendarMode(action.payload)
}
```

Reply:
Good one, this definitely will help in the readability and understatement of the mutations